vercel/resource_project.go

```diff
@@ -1912,16 +1912,14 @@ func (r *projectResource) Update(ctx context.Context, req resource.UpdateRequest
 		})
 	}
 
-	if state.ProtectionBypassForAutomation != plan.ProtectionBypassForAutomation {
-		secret := state.ProtectionBypassForAutomationSecret.ValueString()
-		if plan.ProtectionBypassForAutomationSecret.ValueString() != "" {
-			secret = plan.ProtectionBypassForAutomationSecret.ValueString()
-		}
+	currentSecret := state.ProtectionBypassForAutomationSecret.ValueString()
+	plannedSecret := plan.ProtectionBypassForAutomationSecret.ValueString()
+	if state.ProtectionBypassForAutomation != plan.ProtectionBypassForAutomation && currentSecret != plannedSecret {
```

Comment:
```suggestion
	if state.ProtectionBypassForAutomation != plan.ProtectionBypassForAutomation || currentSecret != plannedSecret {
```

should this be an or? I don't think this actually accomplishes rotating the secret unless `protection_bypass_for_automation` also changes state.

Reply:
:facepalm: Yes, you're absolutely right. Which also implies that this needs to be split up, deleting the secret when it exists but `plan.ProtectionBypassForAutomation` is false.

Reply:
seems like that should be a validation that fails the plan instead of handling at the rpc boundary